aiida_quantumespresso/parsers/neb.py

```diff
@@ -95,168 +94,177 @@ def parse(self, **kwargs):
+        # Now parse the information from the individual pw calculations for the different images
         image_data = {}
         positions = []
         cells = []
+        # for each image...
         for i in range(num_images):
-            # look for xml and parse
-
-            for xml_filename in self._calc.xml_filenames:
-                if xml_filename in list_of_files:
-                    xml_file = os.path.join(out_folder.get_abs_path('.'),
-                                            self._calc._PREFIX +'_{}'.format(i+1),self._calc._PREFIX+'.save',
-                                            xml_filename)
-
+            # check if any of the known XML output file names are present, and parse the first that we find
+            relative_output_folder = os.path.join('{}_{}'.format(PREFIX, i+1), '{}.save'.format(PREFIX))
+            retrieved_files = self.retrieved.list_object_names(relative_output_folder)
+            for xml_filename in PwCalculation.xml_filenames:
+                if xml_filename in retrieved_files:
+                    xml_file_abspath = os.path.join(out_folder._repository._get_base_folder().abspath,
+                                                    relative_output_folder,
+                                                    xml_filename)
                     try:
-                        xml_file_version = get_xml_file_version(xml_file)
-                    except ValueError as exception:
-                        raise QEOutputParsingError('failed to determine XML output file version: {}'.format(exception))
-
-                    if xml_file_version == QeXmlVersion.POST_6_2:
-                        xml_data, structure_dict, bands_data = parse_pw_xml_post_6_2(xml_file)
-                    elif xml_file_version == QeXmlVersion.PRE_6_2:
-                        xml_data, structure_dict, bands_data = parse_pw_xml_pre_6_2(xml_file, dir_with_bands)
-                    else:
-                        raise ValueError('unrecognize XML file version')
-
-                    structure_data = convert_qe2aiida_structure(structure_dict)
+                        with open(xml_file_abspath) as xml_file:
+                            parsed_data_xml, logs_xml = parse_pw_xml(xml_file, None, include_deprecated_v2_keys)
+                    except IOError:
+                        return self.exit_codes.ERROR_OUTPUT_XML_READ
+                    except XMLParseError:
+                        return self.exit_codes.ERROR_OUTPUT_XML_PARSE
+                    except XMLUnsupportedFormatError:
+                        return self.exit_codes.ERROR_OUTPUT_XML_FORMAT
+                    except Exception:
+                        import traceback
+                        traceback.print_exc()
+                        return self.exit_codes.ERROR_UNEXPECTED_PARSER_EXCEPTION
+                    # this image is dealt with, go to next image
+                    break
```

Comment:
This should be a `continue` no? Not a break. With the break it will only ever parse one image

Reply:
No, it breaks out of the inner loop (line 111), the one over the expected XML filenames. If none of the expected filenames are found, the 'break' is never executed and the 'else' clause is triggered.